crates/symbolicator/src/sources.rs

```diff
@@ -144,10 +148,16 @@ pub struct S3SourceKey {
     #[serde(deserialize_with = "deserialize_region")]
     pub region: rusoto_core::Region,
 
+    /// "static" (default)|"container"
+    #[serde(default = "default_aws_provider")]
+    pub aws_credentials_provider: String,
```

Comment:
I think this can be made a custom enum:
```
#[derive(Debug, Clone, Copy, Deserialize, Serialize)]
#[serde(rename_all = "lowercase"]
pub enum AwsCredentialProvider {
    Static,
    Container,
}
```
That way serde will handle the deserialisation and you don't have to panic because the deserialising error is already handled (it needs to be already for e.g. regions).  You can also directly implement `Default` for that and use `#[serde(default)]` on the field in here instead of having a custom default function.

As Arpad mentioned, this config can also come in per-request instead of being statically in the config file (apologies that i didn't fully point this out at first), so panicking is really not something we want to do here.

Reply:
Done, implemented `Default` as suggested, and removed the `panic!`; instead `Static` is now the default provider value.